src/views/Permissions.js

```diff
@@ -0,0 +1,167 @@
+  const logs = await contracts.azimuth.getPastEvents('ChangedKeys', {
+    fromBlock: 0,
+    toBlock: 'latest',
+    filter: { point: [point] },
+  });
+  if (logs.length === 0) {
+    return null;
+  } else {
+    // last log is most recent
+    const block = await web3.eth.getBlock(logs[logs.length - 1].blockNumber);
+    return new Date(block.timestamp * 1000);
+  }
+};
+
+export default function Permissions() {
+  const history = useHistory();
+  const { web3, contracts } = useNetwork();
+  const { wallet } = useWallet();
+  const { pointCursor } = usePointCursor();
+  const { pointCache } = usePointCache();
+
+  const [rekeyDate, setRekeyDate] = useState(Nothing());
+
+  const point = need.point(pointCursor);
+  const pointSize = azimuth.getPointSize(point);
+  const userAddress = need.wallet(wallet).address;
+
+  useLifecycle(() => {
+    getRekeyDate(need.web3(web3), need.contracts(contracts), point).then(
+      res => {
+        if (res === null) setRekeyDate(Just('unknown'));
+        else setRekeyDate(Just(res.toString()));
+      }
+    );
+  });
+
+  const goSetManagement = useCallback(
+    () => history.push(ROUTE_NAMES.SET_MANAGEMENT_PROXY),
+    [history]
+  );
+
+  const goSetSpawn = useCallback(
+    () => history.push(ROUTE_NAMES.SET_SPAWN_PROXY),
+    [history]
+  );
+
+  const goSetVoting = useCallback(
+    () => history.push(ROUTE_NAMES.SET_VOTING_PROXY),
+    [history]
+  );
+
+  const goSetKeys = useCallback(() => history.push(ROUTE_NAMES.SET_KEYS), [
+    history,
+  ]);
+
+  const pointDetails = need.fromPointCache(pointCache, point);
+
+  const isOwner = eqAddr(userAddress, pointDetails.owner);
+  const canManage =
+    isOwner || eqAddr(userAddress, pointDetails.managementProxy);
+
+  const proxyAction = (name, address, onClick) => {
+    if (address === '0x0000000000000000000000000000000000000000')
+      address = 'Unset';
+    return (
+      <Grid.Item full>
+        <ForwardButton disabled={!isOwner} onClick={onClick} detail={address}>
+          {`${name} proxy address`}
+        </ForwardButton>
+      </Grid.Item>
+    );
+  };
+
+  let items = 3;
+
+  let spawnProxyAction = null;
+  if (pointSize !== azimuth.PointSize.Planet) {
+    items++;
+    spawnProxyAction = proxyAction(
+      'Spawn',
+      pointDetails.spawnProxy,
+      goSetSpawn
+    );
+  }
+
+  let votingProxyAction = null;
+  if (pointSize === azimuth.PointSize.Galaxy) {
+    items++;
+    votingProxyAction = proxyAction(
+      'Voting',
+      pointDetails.votingProxy,
+      goSetVoting
+    );
+  }
+
+  const netKeysText =
+    pointDetails.keyRevisionNumber === 0 ? (
+      'Not yet configured'
+    ) : (
+      <>
+        {`Revision #${pointDetails.keyRevisionNumber}, since `}
+        {Just.hasInstance(rekeyDate) ? (
+          rekeyDate.value
+        ) : (
+          <Blinky a={LOADING_CHARACTER} b={INTERSTITIAL_CHARACTER} delayed />
+        )}
+      </>
+    );
+
+  return (
+    <View>
+      <Grid className={`pt${items}`}>
```

Comment:
why the dynamic padding here, and why is it based on the number of items?

Reply:
Because I have no idea what I'm doing. (^:

I had seen a couple Grids with `pt` numbers equal to their item count, and figured there was a relation there. There's... not, hah.

Reply:
what was the prior art you were looking at, for context?

Reply:
Might've been Point? Unsure...

Reply:
¯\\\_(ツ)_/¯